.github/actions/setup-build/action.yml

```diff
@@ -0,0 +1,120 @@
+name: "Cached build"
+description: "Setup the build using cache"
+inputs:
+  ghc:
+    description: "Ghc version"
+    required: true
+  cabal:
+    description: "Cabal version"
+    required: false
+    default: "3.6"
+  os:
+    description: "Operating system: Linux, Windows or macOS"
```

Comment:
Interesting. Seems like you discovered we fully abstracted from `ubuntu-latest`-kind of values into `runner.os`.

It would be such until installation & run would depend on OS versions, lets hope it would be invariant for a long time.